services/web/server/src/simcore_service_webserver/login/handlers.py

```diff
@@ -129,27 +215,65 @@ async def login(request: web.Request):
     user = await db.get_user({"email": email})
     if not user:
         raise web.HTTPUnauthorized(
-            reason=cfg.MSG_UNKNOWN_EMAIL, content_type="application/json"
+            reason=cfg.MSG_UNKNOWN_EMAIL, content_type=MIMETYPE_APPLICATION_JSON
         )
 
     if user["status"] == BANNED or user["role"] == ANONYMOUS:
         raise web.HTTPUnauthorized(
-            reason=cfg.MSG_USER_BANNED, content_type="application/json"
+            reason=cfg.MSG_USER_BANNED, content_type=MIMETYPE_APPLICATION_JSON
         )
 
     if not check_password(password, user["password_hash"]):
         raise web.HTTPUnauthorized(
-            reason=cfg.MSG_WRONG_PASSWORD, content_type="application/json"
+            reason=cfg.MSG_WRONG_PASSWORD, content_type=MIMETYPE_APPLICATION_JSON
         )
 
     if user["status"] == CONFIRMATION_PENDING:
         raise web.HTTPUnauthorized(
-            reason=cfg.MSG_ACTIVATION_REQUIRED, content_type="application/json"
+            reason=cfg.MSG_ACTIVATION_REQUIRED, content_type=MIMETYPE_APPLICATION_JSON
         )
 
     assert user["status"] == ACTIVE, "db corrupted. Invalid status"  # nosec
     assert user["email"] == email, "db corrupted. Invalid email"  # nosec
 
+    if settings.LOGIN_2FA_REQUIRED:
+        if not user["phone"]:
+            response = web.json_response(
+                {
+                    "data": {
+                        "code": "PHONE_NUMBER_REQUIRED",  # this string is used by the frontend
+                        "reason": "PHONE_NUMBER_REQUIRED",
+                    },
+                    "error": None,
+                },
+                status=web.HTTPAccepted.status_code,
+            )
+            return response
+
+        assert user["phone"]  # nosec
```

Comment:
Since this can be None, could you check and raise an error here in that case.
Ideally there should be a validator for the phone number as well. But just a the above check is already better.

Reply:
it should NOT be None in this context. That is why i added assertion, to make sure the external context respects this